django/core/files/move.py

```diff
@@ -40,8 +26,11 @@ def file_move_safe(
     ``FileExistsError``.
     """
     # There's no reason to move if we don't have to.
-    if _samefile(old_file_name, new_file_name):
-        return
+    try:
+        if os.path.samefile(old_file_name, new_file_name):
+            return
+    except FileNotFoundError:
```

Comment:
I think we should catch all `OSError`s to keep the current behavior, e.g. `PermissionError` may happen.